pkg/proof/proof.go

```diff
@@ -103,7 +104,17 @@ func GenerateZkProof(ctx context.Context, circuitPath string, inputs ZKInputs) (
 	}
 
 	// generate proof
-	proveCmd := exec.Command("snarkjs", "groth16", "prove", circuitPath+"/circuit_final.zkey", wtnsFile.Name(), proofFile.Name(), publicFile.Name())
+	var execCommandName string
+	var execCommandParams []string
+	if useRapidsnark {
+		execCommandName = "prover"
+	} else {
+		execCommandName = "snarkjs"
+		execCommandParams = append(execCommandParams, "groth16", "prove")
+	}
+	execCommandParams = append(execCommandParams, circuitPath+"/circuit_final.zkey", wtnsFile.Name(), proofFile.Name(), publicFile.Name())
+	proveCmd := exec.Command(execCommandName, execCommandParams...)
+	log.WithContext(ctx).Debugf("used prover: %s", execCommandName)
```

Comment:
if we do the generation of proof with rapidsnark should we also verify with rapidsnark library? As I understood snarkjs can be not installed in a container in case of using rapidsnark

Reply:
No, rapidsnark provides only proof generation functionality, without verification.